src/machine/image-varlink.c

```diff
@@ -0,0 +1,84 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-json.h"
+#include "sd-varlink.h"
+
+#include "bus-polkit.h"
+#include "image-varlink.h"
+#include "machine.h"
+#include "string-util.h"
+
+int vl_method_update_image(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        struct params {
+                const char *image_name;
+                const char *new_name;
+                int read_only;
+                uint64_t limit;
+        };
+
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "name",     SD_JSON_VARIANT_STRING,        sd_json_dispatch_const_string, offsetof(struct params, image_name), SD_JSON_MANDATORY },
```

Comment:
the other calls also allowed referencing the machine by pid. I think just to be systematic we should retain that here, even if that doesn't make too much sense to actually IRL.

Reply:
here I'm referencing image, not machine. Does it make sense to reference by pid?

Reply:
oh, right, ignore me. i am an idiot